feat: add badge to sidebar link (#516)
withastro/starlight: docs/src/content/docs/reference/frontmatter.md

### `sidebar`

> **type:** `{ label?: string; order?: number; hidden?: boolean, badge?: string | { text: string; variant: 'blue' | 'pink' | 'green' | 'yellow' | 'purple'; } }`

Author: ```suggestion
**type:** `{ label?: string; order?: number; hidden?: boolean, badge?: string | { text: string; variant: 'blue' | 'pink' | 'green' | 'yellow' | 'purple' | 'default' ; } }`
```

  hidden: true
---
```

#### `badge`

> **type:** `string | { text: string; variant: 'blue' | 'pink' | 'green' | 'yellow' | 'purple'; }`

Author: ```suggestion
**type:** `string | { text: string; variant: 'blue' | 'pink' | 'green' | 'yellow' | 'purple'; | 'default' }`
```

Add a badge to the page in the sidebar when displayed in an autogenerated group of links.

> When using a string, the badge will be displayed with a default `blue` variant.

Author: ```suggestion
When using a string, the badge will be displayed with a default accent color.
```
